source/menu/m_menu.qc

```diff
@@ -550,6 +552,146 @@ void(float order, vector minmaxsteps, string cvar_s, float is_int, float no_text
+{
+	float maxlen = 0;
+
+	if (getTextWidth(text, MENU_TEXT_SMALL.x) + 18 >= size.x)
+		maxlen = strlen(text);
+	else
+		maxlen = 128;
+
+	sui_action_element(pos, size, id, sui_noop);
+	if (sui_is_clicked(id)) cursor = strlen(text);
+	if (sui_is_hovered(id))
+	{
+		float char = 0;
+		float scan = 0;
+		while(sui_get_input(char, scan)) sui_handle_text_input(char, scan, maxlen, text, cursor);
+	}
+};
+
+static void(string id, vector pos, vector size, string text, float cursor) draw_text_input =
+{
+	sui_push_frame(pos - [6, 5], size);
+	sui_fill([0, 0], size, [0, 0, 0], 1.0, 0);
+	sui_border_box([0, 0], size, 2, [0.2, 0.2, 0.2], 0.3, 0);
+	sui_set_align([SUI_ALIGN_START, SUI_ALIGN_CENTER]);
+	sui_text([6, 0], MENU_TEXT_SMALL, text, MENU_TEXT_1, 1, 0);
+	float cursor_x = getTextWidth(substring(text, 0, cursor), MENU_TEXT_SMALL.x);
+	if (sui_is_hovered(id))
+		sui_text([6 + cursor_x, 0], MENU_TEXT_SMALL, "|", [1.0, 0.25, 0.25], 1, 0);
+	sui_pop_frame();
+};
+
+//
+// Menu_TextInput(id, order, text, cursor)
+// Draws a Text Input box at the given order position
+//
+void(string id, float order, __inout string text, __inout float cursor) Menu_TextInput =
+{
+	vector pos = [320, 50 + (order * 25)];
+	vector size = [290, 22];
+
+	handle_text_input(id, pos, size, text, cursor);
+
+	draw_text_input(id, pos, size, text, cursor);
+};
+
+//
+// Menu_PasswordInput(id, order, text, cursor)
+// Draws a Text Input box at the given order position (blocked out with asterisks)
+//
+void(string id, float order, __inout string text, __inout float cursor) Menu_PasswordInput =
+{
+	vector pos = [320, 50 + (order * 25)];
+	vector size = [290, 22];
+	string safe = "";
+
+	for (int i = 0; i < strlen(text); i++)
+		safe = sprintf("%s%s", safe, "*");
+
+	handle_text_input(id, pos, size, text, cursor);
+
+	draw_text_input(id, pos, size, safe, cursor);
+};
+
+static string(string s) str2ascii =
+{
+	for (int i = 0; i < strlen(s); i++)
+	{
+		if (s[i] < 32 || s[i] > 126)
+			s[i] = 63; // question mark
+	}
+	return s;
+};
+
+//
+// Menu_ServerList(id, pos, size, scrollofs, num_servers)
+// Draw the master server list
+//
+void(string id, vector pos, vector size, __inout vector scrollofs, float num_servers) Menu_ServerList =
+{
+	vector rowsize = [size.x, 16];
+	vector listitem_pos = [0, 0, 0];
+	sui_fill(pos, size, [0, 0, 0], 0.75, 0);
+	sui_list_view_begin(strcat(id, "scrl"), pos, size, rowsize, num_servers, scrollofs, [0, 16]);
+	for (float index = sui_list_item(listitem_pos); index > -1; index = sui_list_item(listitem_pos))
+	{
+		sui_push_frame(listitem_pos, rowsize);
+		sui_set_align([SUI_ALIGN_START, SUI_ALIGN_CENTER]);
+
+		string listitem_id = strcat(id, "scrl", ftos(index));
+		sui_action_element([0, 0], rowsize, listitem_id, sui_noop);
+
+		// bg
+		vector bpos = [0, 0];
+		vector bsize = [(rowsize.x / 8) * 5, rowsize.y];
+		sui_fill(bpos, rowsize, [0.25, 0.25, 0.25], 0.5, 0);
```

Comment:
The menu here is very gray, really not a lot of color at all. Could you add a red hue to this, or maybe surround the widget in a yellow box or something? Just to add some color

Reply:
Addressed in [854530b](https://github.com/nzp-team/quakec/pull/90/commits/854530bf8a01b7afe1eb3cc7bdcc1a1226f7c2a4)